ra2ce/graph/networks.py

```diff
@@ -202,43 +206,45 @@ def check_base_file(file_type: str, file_path: Path):
         self.base_network_crs = _network_gdf.crs
         return _base_graph, _network_gdf
 
-    def create(self) -> dict:
+    def create(self) -> GraphFiles:
         """Handler function with the logic to call the right functions to create a network.
 
         Returns:
             (dict): A dict of a network (GeoDataFrame) and 1 (base NetworkX graph) or 2 graphs (base NetworkX and OD graph)
         """
         # Save the 'base' network as gpickle and if the user requested, also as shapefile.
-        to_save = ["pickle"] if not self._network_config.save_gpkg else ["pickle", "shp"]
+        to_save = (
+            ["pickle"] if not self._network_config.save_gpkg else ["pickle", "shp"]
+        )
         od_graph = None
         base_graph = None
         network_gdf = None
 
         # For all graph and networks - check if it exists, otherwise, make the graph and/or network.
-        if not (self.files["base_graph"] or self.files["base_network"]):
+        if not (self.files.base_graph or self.files.base_network):
             base_graph, network_gdf = self._get_new_network_and_graph(to_save)
         else:
             base_graph, network_gdf = self._get_stored_network_and_graph(
-                self.files["base_graph"], self.files["base_network"]
+                self.files.base_graph, self.files.base_network
             )
 
         # create origins destinations graph
         if (
             (self.origins)
             and (self.destinations)
-            and not self.files["origins_destinations_graph"]
+            and not self.files.origins_destinations_graph
         ):
             # reading the base graphs
-            if self.files["base_graph"] and base_graph:
-                base_graph = GraphPickleReader().read(self.files["base_graph"])
+            if self.files.base_graph and base_graph:
+                base_graph = GraphPickleReader().read(self.files.base_graph)
             # adding OD nodes
             if self.origins.suffix == ".tif":
                 self.origins = self.generate_origins_from_raster()
             od_graph = self.add_od_nodes(base_graph, self.base_graph_crs)
             self._export_network_files(od_graph, "origins_destinations_graph", to_save)
 
-        return {
-            "base_graph": base_graph,
-            "base_network": network_gdf,
-            "origins_destinations_graph": od_graph,
-        }
+        return GraphFiles(
```

Comment:
Here we are not setting paths, but rather graphs, which means the type hints are not correct.

Reply:
Rebuilt